Sming/Core/Network/MqttClient.cpp

```diff
@@ -225,6 +225,11 @@ bool MqttClient::connect(const Url& url, const String& clientName, uint32_t sslO
 		}
 	}
 
+	if(requestQueue.full()) {
```

Comment:
The suggested change is improvement to what we had but I was wondering if we can improve it further without too much overhead.
What I mean is the following:  While piling X normal publish messages in the queue is a good idea piling X connect messages is not. So my suggestion would be to look in the queue for connect message and if found replace it with the new connect message. A connect message should be the first to be dequeued. There should be max one connect message in the queue.